### DISCUSSION TITLE
With Visual Studio support dropped in GitVersion.MSBuild 6.0.0 are there any plans for accessing 5.12.0 documentation?

### DISCUSSION THREAD

**Discussion by RobSmyth** [GitTools/GitVersion#4169] [General]
Hi,

With Visual Studio support being dropped in GitVersion.MSBuild 6.0.0 it means that .NET developers using GitVersion.MSBuild, and needing dev build versioning, need to stay on 5.12.0. The 5.12.0 configuration is different and it seems to no longer be available on the [documentation site](https://gitversion.net/docs/reference/configuration). Are there plans to provide 5.12.0 documentation to support for those continuing to use _any_ GitVersion 5.x products?

Another thing, now that the documentation site updated to 6.0.0 (nice) I did not see anything obvious saying that the configuration shown is for 6.0.0 and later and different to 5.x configuration. So those with working 5.x installs, oblivious to 6.0.0, going to the site to maintain their project (using any GitVersion product) may not have a good experience.

Plans moving forward?

**Comment by arturcic** [comment]
@RobSmyth, that's a valid point. I will check what are the options for hosting version 5.12. We need to also be explicit which version is on the main docs and have a reference to the previous version. That might take some time to implement, but that's doable. Thanks for pointing out.

**Reply by arturcic** [reply]
@RobSmyth I added the `5.12.0` docs at https://gitversion.net/5.12.0/docs/

**Reply by RobSmyth** [reply]
Thanks

**Comment by RevZero** [comment]
Forgive my confusion, but I can't find any mention about Visual Studio support having been dropped with v6. Can you point me to this information? 
I just recently updated to v6 and noticed it wasn't working properly. Shouldn't this be mentioned at https://gitversion.net/docs/usage/msbuild?

Also, could someone please explain to me, what is the actual use case for the GitVersion.MSBuild package if not for Visual Studio? Wouldn't other IDEs / Platform use "dotnet build" anyway?

**Reply by arturcic** [reply]
https://github.com/GitTools/GitVersion/pull/4229 - I think we will merge this for now. I still hope we can find a solution how to bring the support back.

**Reply by RobSmyth** [reply]
> Also, could someone please explain to me, what is the actual use case for the GitVersion.MSBuild package if not for Visual Studio? Wouldn't other IDEs / Platform use "dotnet build" anyway?

dotnet CLI also uses MSBuild but another version of it than what VS uses. So using GitVersion.MSBuild your dotnet CLI builds will have versioning without the need for any external scripts or build steps. Nice and simple, it means that devs, who do not use Visual Studio, also potentially get versioning on their builds. Environmental parity.

The need for external versioning (another build step, script, or tool like GitVersion CLI) may come when you have other, non-.NET, languages in the project or you need to customise the generated version information (further scripting).

In release 5, GitVersion.MSBuild was built against .NET standard 2.0 and therefore usable by all long term support .NET releases. Note that .NET Framework 4.8.1 is a long term supported release, for Windows, but there will not be further releases.

Microsoft:

> .NET Framework 4.8.1 is the latest version of .NET Framework and will continue to be distributed with future releases of Windows. As long as it is installed on a supported version of Windows, .NET Framework 4.8.1 will continue to also be supported.
